category/basic.lean

```diff
@@ -44,14 +58,33 @@ end
   (g <$> (x <*> y)) = ((∘) g) <$> x <*> y :=
 by simp [(pure_seq_eq_map _ _).symm]; simp [seq_assoc]
 
+private def mpartition_aux (x : α) : ulift bool → list α × list α → list α × list α
+| ⟨tt⟩ (xs,ys) := (x::xs,ys)
+| ⟨ff⟩ (xs,ys) := (xs,x::ys)
+
+def list.mpartition' (g : α → f (ulift bool)) : list α → f (list α × list α)
+| [] := pure ([],[])
+| (x :: xs) := mpartition_aux x <$> g x <*> list.mpartition' xs
+
+def list.mpartition {α : Type} {f : Type → Type v} [applicative f] [is_lawful_applicative f]
+  (g : α → f bool) :=
+list.mpartition' (λ x, ulift.up <$> g x)
+
+variables {m : Type u → Type v} [applicative m]
+def lift₂
```

Comment:
I think we prefer to directly use `f <$> ma₁ <*> ma₂`

Reply:
I just noticed that I don't it in this patch. I'll take it out.